audio/buffer_source_node.rs

```diff
@@ -61,12 +70,19 @@ pub(crate) struct AudioBufferSourceNode {
     /// An playhead position where looping should begin if the loop_enabled
     /// attribute is true.
     loop_start: Option<f64>,
-    /// Playback offset.
-    playback_offset: usize,
-    /// The speed at which to render the audio stream.
+    /// The speed at which to render the audio stream. Can be negative if the
+    /// audio is to be played backwards. With a negative playback_rate, looping
+    /// jumps from loop_start to loop_end instead of the other way around.
     playback_rate: Param,
     /// Time at which the source should start playing.
     start_at: Option<Tick>,
+    /// Offset parameter passed to Start().
+    start_offset: Option<f64>,
+    /// Duration parameter passed to Start().
+    start_duration: Option<f64>,
+    /// The same as start_at, but with subsample accuracy.
+    /// FIXME: Maybe AudioScheduledSourceNode should use this as well?
+    start_when: f64,
```

Comment:
Yes, please. Let's make this part of `AudioScheduledSourceNode`. It could be done in a follow-up though.